src/qt/metadexdialog.cpp

```diff
@@ -530,14 +529,36 @@ void MetaDExDialog::recalcTotal(bool useBuyFields)
         amount = StrToInt64(ui->sellAmountLE->text().toStdString(),divisible);
         price = StrToInt64(ui->sellPriceLE->text().toStdString(),true);
     }
-    totalPrice = amount * price;
 
-    // error and overflow detection
-    if (0 >= amount || 0 >= price || totalPrice < amount || totalPrice < price || (totalPrice / amount != price)) {
-        if (useBuyFields) { ui->buyTotalLabel->setText("N/A"); return; } else { ui->sellTotalLabel->setText("N/A"); return; }
+    if (0 >= amount || 0 >= price) {
+        if (useBuyFields) {
+            ui->buyTotalLabel->setText("N/A");
+        } else {
+            ui->sellTotalLabel->setText("N/A");
+        }
+        return;
+    }
+
+    uint256 amount256 = ConvertTo256(amount);
+    uint256 price256 = ConvertTo256(price);
+    uint256 totalPrice256 = amount256 * price256;
+    if (divisible) totalPrice256 = totalPrice256 / COIN;
+
+    if (totalPrice256 > std::numeric_limits<int64_t>::max()) {
```

Comment:
This may work only one way (e.g. `uint256 > int64_t`, but not `int64_t < uint256`), better use `uint256_const::max_int64` (also in [uint256_extensions.h](https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/src/omnicore/uint256_extensions.h#L22)) instead of `std::numeric_limits<int64_t>::max()`.